src/pp-annotator/api/DebugStorage.ts

```diff
@@ -24,36 +24,38 @@ export default class DebugStorage implements IStorage, IModule {
     }
   }
 
-  create(annotation: IAnnotation): IAnnotation {
+  create(annotation: IAnnotationAPIModel): IAnnotationAPIModel {
     if (!annotation.id) {
       annotation.id = generateId();
     }
+    annotation.create_date = new Date(Date.now()).toISOString();
+    annotation.does_belong_to_user = true;
     this.trace('create', annotation);
 
     this.annotations = this.annotations.concat(annotation);
     return annotation;
   }
 
-  update(annotation: IAnnotation): IAnnotation {
+  update(annotation: IAnnotationAPIModel): IAnnotationAPIModel {
     this.trace('update', annotation);
 
     this.delete(annotation);
     return this.create(annotation);
   }
 
-  delete(annotation: IAnnotation): IAnnotation {
+  delete(annotation: IAnnotationAPIModel): IAnnotationAPIModel {
     this.trace('delete', annotation);
-
     const deleted = this.query(annotation.id);
     this.annotations = this.annotations.filter(ann => ann.id !== annotation.id);
-
-    return deleted;
+    return deleted[0];
   }
 
-  query(id: string | number): IAnnotation {
-    this.trace('query', id);
-
-    return this.annotations.find(annotation => annotation.id === id) as IAnnotation;
+  query(id?: number): IAnnotationAPIModel[] {
```

Comment:
why the change?

Reply:
If you mean the `id`'s optionality in `query` method, the reason is to provide an intuitive way to get all/selected annotations.
If you mean the overall change: I needed to give up on the `DebugStorage's` universality and simulate logic on an annotation's creation (author saving + creation date). In the same vein, I gave up on `string | number` alternative (I knew we would not provide a string any more and didn't want to maintain that).